File: C++/1110-더하기 사이클.cpp
#include <iostream>
using namespace std;

int main() {
    int ogn, chg, a, b, c, cycle = 0;
    cin >> ogn;
    int tmp = ogn;

    while (1) {
        a = tmp/10, b = tmp%10, c = a + b;
        chg = b * 10 + c % 10;
        cycle++;
        if (ogn == chg) {
            break;
        }
        else {
            tmp = chg;
        }
    }
    cout << cycle;
    return 0;
}

File: Java/1110-더하기 사이클.java
import java.util.*;

class Main{
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int ogn = sc.nextInt();
                int cycle = 0, tmp = ogn;
        
        while (true) {
            int a = tmp/10, b = tmp%10, c = a + b;
            int chg = b * 10 + c % 10;
            cycle++;
            if (ogn == chg) {
                break;
            }
            else {
                tmp = chg;
            }
        }
        System.out.println(cycle);
    }
}
